Source/WebCore/platform/graphics/MediaPlayerPrivate.h

```diff
@@ -125,15 +125,16 @@ class MediaPlayerPrivateInterface {
     virtual double durationDouble() const { return duration(); }
     virtual MediaTime durationMediaTime() const { return MediaTime::createWithDouble(durationDouble()); }
 
-    virtual float currentTime() const { return 0; }
-    virtual double currentTimeDouble() const { return currentTime(); }
-    virtual MediaTime currentMediaTime() const { return MediaTime::createWithDouble(currentTimeDouble()); }
+    WEBCORE_EXPORT MediaTime apparentMediaTime() const;
+    virtual MediaTime currentMediaTime() const { return MediaTime::zeroTime(); }
     virtual bool currentMediaTimeMayProgress() const { return readyState() >= MediaPlayer::ReadyState::HaveFutureData; }
 
     virtual bool setCurrentTimeDidChangeCallback(MediaPlayer::CurrentTimeDidChangeCallback&&) { return false; }
 
     virtual MediaTime getStartDate() const { return MediaTime::createWithDouble(std::numeric_limits<double>::quiet_NaN()); }
 
+    virtual void aboutToSeek(const MediaTime& time) { m_aboutToSeekTime = time; }
+    virtual MediaTime aboutToSeekTime() const { return m_aboutToSeekTime; }
```

Comment:
Nit: `aboutToSeek` doesn't seem right. Maybe `willSeekToTarget` and `pendingSeekTime` instead?

Reply:
there were other methods named `aboutTo...` so I used that.

tbh, I changed the name of that method 3 times before this draft.